app/integrations/comp_and_pen/scheduled_message_helpers.py

```diff
@@ -0,0 +1,207 @@
+import boto3
+from boto3.dynamodb.conditions import Attr
+from flask import current_app
+
+from app.models import (
+    SMS_TYPE,
+    Service,
+    Template,
+)
+from app.notifications.send_notifications import send_notification_bypass_route
+from app.va.identifier import IdentifierType
+
+
+class CompPenMsgHelper:
+    dynamodb_table = None
+
+    def __init__(self, dynamodb_table_name: str) -> None:
+        """
+        This class is a collection of helper methods to facilitate the delivery of schedule Comp and Pen notifications.
+
+        :param dynamodb_table_name (str): the name of the dynamodb table for the db operations, required
+        """
+        self.dynamodb_table_name = dynamodb_table_name
+
+    def _connect_to_dynamodb(self, dynamodb_table_name: str = None) -> None:
+        """Establishes a connection to the dynamodb table with the given name.
+
+        :param dynamodb_table_name (str): the name of the dynamodb table to establish a connection with
+
+        Raises:
+            ClientError: if it has trouble connectingto the dynamodb
+        """
+        if dynamodb_table_name is None:
+            dynamodb_table_name = self.dynamodb_table_name
+
+        # connect to dynamodb table
+        dynamodb_resource = boto3.resource('dynamodb')
+        self.dynamodb_table = dynamodb_resource.Table(dynamodb_table_name)
+
+    def get_dynamodb_comp_pen_messages(
+        self,
+        message_limit: int,
+    ) -> list:
+        """
+        Helper function to get the Comp and Pen data from our dynamodb cache table.
+
+        Items should be returned if all of these attribute conditions are met:
+            1) item exists on the `is-processed-index`
+            2) payment_id is not equal to -1 (placeholder value)
+            3) paymentAmount exists
+
+        :param message_limit: the number of rows to search at a time and the max number of items that should be returned
+        :return: a list of entries from the table that have not been processed yet
+
+        https://boto3.amazonaws.com/v1/documentation/api/latest/guide/dynamodb.html#querying-and-scanning
+        """
+
+        if self.dynamodb_table is None:
+            self._connect_to_dynamodb()
+
+        is_processed_index = 'is-processed-index'
+
+        filters = Attr('payment_id').exists() & Attr('payment_id').ne(-1) & Attr('paymentAmount').exists()
```

Comment:
payment_id will never be equal to -1 now that the consumer is the one inserting items into dynamodb

Reply:
I removed the attr referencing 'payment_id' since we don't have to worry about that anymore.